dms/models/ir_attachment.py

```diff
@@ -36,8 +36,21 @@ def _dms_directories_create(self):
             )
 
     def _dms_operations(self):
+        """Perform the operation only if there is a storage with linked models.
+        The directory (dms.directory) linked to the record (if it does not exist)
+        and the file (dms.file) with the linked attachment would be created.
+        """
+        # Apply sudo to prevent ir.rule from being applied.
+        storages = self.env["dms.storage"].sudo().search([("model_ids", "!=", False)])
+        dms_models = storages.mapped("model_ids.model")
```

Comment:
If we continue this way, this must be cached and invalidate when changes. But I don't ge why `res_model` and `res_id`, which is a core mechanism, is filled for the DMS, and we may need to avoid it and use other fields.